core/src/main/scala/org/genivi/sota/core/CampaignResource.scala

```diff
@@ -47,9 +50,17 @@ class CampaignResource(namespaceExtractor: Directive1[AuthedNamespaceScope],
     complete(db.run(Campaigns.fetch(id)))
   }
 
-  def launch(id: Campaign.Id, lc: LaunchCampaign): Route = {
-    lc.isValid match {
-      case Xor.Right(()) => complete(CampaignLauncher.launch(deviceRegistry, updateService, id, lc, messageBus))
+  def launch(id: Campaign.Id, lc: LaunchCampaignRequest): Route = {
+    lc.isValid() match {
+      case Xor.Right(()) =>
+        val r = async {
+          val campaign = await(db.run(Campaigns.fetch(id)))
+          campaign.meta.deltaFrom match {
+            case Some(_) => await(CampaignLauncher.generateDeltaRequest(id, lc, messageBus))
+            case None => await(CampaignLauncher.launch(deviceRegistry, updateService, id, lc, messageBus))
+          }
+        }
+        complete(NoContent -> r)
```

Comment:
I am not too sure this will work properly. `r` is a `Future[_]`, not sure this will wait on the future before returning.

This should be a `Future[StatusCode]` returning an instance of `NoContent`.